repos/system_upgrade/el7toel8/actors/kernel/checkinstalleddevelkernels/libraries/checkinstalleddevelkernels.py

```diff
@@ -27,8 +27,8 @@ def process():
         summary = ('DNF cannot produce a valid upgrade transaction when'
                    ' multiple kernel-devel packages are installed.')
         hint = ('Remove all but one kernel-devel packages before running Leapp again.')
-        commands = ([['yum', '-y', 'remove', '{n}-{v}.{r}'.format(
-            n=pkg.name, v=pkg.version, r=pkg.release)] for pkg in pkgs[:-1]])
+        packages = ['{n}-{v}.{r}'.format(n=pkg.name, v=pkg.version, r=pkg.release) for pkg in pkgs[:-1]]
+        commands = [['yum', '-y', 'remove'] + packages]
```

Comment:
@shaded-enmity This is wrong too. This will lead in sutation you remove all devel-kernels. But you are supposed to pick one that should be kept and remove the others. This command does a different think.

Reply:
last change made it clear completely.